### PR TITLE
Improvements to app documentation

### PR DESCRIPTION
**Pull request by mikewilburn**
This PR:
- renames `docs/index.md` to `docs/README.md`
- adds table of contents to (2) README files. The TOC was generated using MDTOC.
- links to `docs/README.md` from the root README.md
- makes small formatting corrections to root README.md
- increments app version to the applicable patch number
- describes patch changes in RELEASE.md

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Release 1.0.2
```
Comment (esreli) [inline RELEASE.md:1]: We may not have these headers consistent from one OSA repo to another but I do think they should remain consistent within each project.
```suggestion
## 1.0.2
```

File: docs/README.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
+# Nearby Places JavaScript
```
Comment (esreli) [inline docs/README.md:1]: The convention has been to name this header 'Contents'. Should we keep doing so?
```suggestion
# Contents
```

File: RELEASE.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+
+- Adds doc table of contents to root README.md and docs/index.md
+- Renames docs/index.md to [docs/README.md](/docs/README.md)
+
 ## 1.0.1
 
-- Comprehensive [app documentation](/documentation/index.md) from the ArcGIS for Developers site.
\ No newline at end of file
+- Comprehensive [app documentation](/documentation/index.md) from the ArcGIS for Developers site.
```
Comment (esreli) [inline RELEASE.md:8]:
```suggestion
- Comprehensive [app documentation](/docs/README.md) from the ArcGIS for Developers site.
```

**Review by esreli — CHANGES_REQUESTED** [review]
Just a few changes :)

File: RELEASE.md (hunk 1 above)
Comment (mikewilburn) [reply RELEASE.md:1]: Very good catch!

File: docs/README.md (hunk 2 above)
Comment (mikewilburn) [reply docs/README.md:1]: I don't think we have a single convention. I've been placing a "Contents" header or emboldening one above the TOC when the document has an introductory paragraph and it feels the TOC is better placed after that. In cases such as those, I think a TOC title helps, but when it's immediately at the top of the document, I think the document header naming the app is fine. I'd like to leave this as it is.

**Comment by mikewilburn** [comment]
@esreli good eyes on the RELEASE doc. I just pushed up fixes to correct those two things. Let me know what you think about my reply to the TOC header discussion.

File: RELEASE.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## 1.0.2
+
+- Adds doc table of contents to root README.md and docs/index.md
+- Renames docs/index.md to [docs/README.md](/docs/README.md)
+
 ## 1.0.1
 
-- Comprehensive [app documentation](/documentation/index.md) from the ArcGIS for Developers site.
\ No newline at end of file
+- Comprehensive [app documentation](/documentation/README.md) from the ArcGIS for Developers site.
```
Comment (esreli) [inline RELEASE.md:8]:
```suggestion
- Comprehensive [app documentation](/docs/README.md) from the ArcGIS for Developers site.
```

File: docs/README.md (hunk 2 above)
Comment (esreli) [reply docs/README.md:1]: Sure 👍

**Review by esreli — CHANGES_REQUESTED** [review]
One more docs link fix and I think we're good to go.

File: RELEASE.md (hunk 4 above)
Comment (mikewilburn) [reply RELEASE.md]: Oof, don't know how I missed that. Thanks, just corrected.

**Review by esreli — APPROVED** [review]

**Comment by mikewilburn** [comment]
@odoe do you have any comments on this PR? In particular, can you confirm for me that ea39852 caught the one place in which to update the app version?

**Review by odoe — APPROVED** [review]

**Pull request merged**